src/backend/marsha/core/tests/test_api_liveregistration.py

```diff
@@ -1997,9 +2092,620 @@ def test_api_liveregistration_create_role_none_email_empty(self):
+        video = VideoFactory()
+        liveregistration = LiveRegistrationFactory(
+            consumer_site=video.playlist.consumer_site,
+            email=None,
+            is_registered=False,
+            live_attendance={"key1": {"sound": "OFF", "tabs": "OFF"}},
+            lti_user_id="56255f3807599c377bf0e5bf072359fd",
+            lti_id="Maths",
+            video=video,
+        )
+        self.assertEqual(LiveRegistration.objects.count(), 1)
+        jwt_token = AccessToken()
+        jwt_token.payload["context_id"] = "Maths"
+        jwt_token.payload["consumer_site"] = str(video.playlist.consumer_site.id)
+        jwt_token.payload["resource_id"] = str(video.id)
+        jwt_token.payload["roles"] = [
+            random.choice(["administrator", "instructor", "student", ""])
+        ]
+        jwt_token.payload["user"] = {
+            "email": "chantal@aol.com",
+            "id": "56255f3807599c377bf0e5bf072359fd",
+            "username": "Token",
+        }
+        timestamp = str(timezone.now())
+        response = self.client.post(
+            "/api/liveregistrations/push_attendance/",
+            {"live_attendance": {timestamp: {"sound": "ON", "tabs": "OFF"}}},
+            content_type="application/json",
+            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+        )
+        liveregistration.refresh_from_db()
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            json.loads(response.content),
+            {
+                "id": str(liveregistration.id),
+                "live_attendance": {
+                    "key1": {"sound": "OFF", "tabs": "OFF"},
+                    timestamp: {"sound": "ON", "tabs": "OFF"},
+                },
+                "video": str(video.id),
+            },
+        )
+        # no new object has been created
+        self.assertEqual(LiveRegistration.objects.count(), 1)
+        # update username and email with current token
+        self.assertEqual(liveregistration.email, "chantal@aol.com")
+        self.assertEqual(liveregistration.username, "Token")
+        self.assertEqual(
+            liveregistration.live_attendance,
+            {
+                "key1": {"sound": "OFF", "tabs": "OFF"},
+                timestamp: {"sound": "ON", "tabs": "OFF"},
+            },
+        )
+
+    def test_api_liveregistration_post_attendance_token_public(
+        self,
+    ):
+        """Can't create attendance when it's not a LTI token"""
+        video = VideoFactory()
+        self.assertEqual(LiveRegistration.objects.count(), 0)
+        jwt_token = AccessToken()
+        jwt_token.payload["resource_id"] = str(video.id)
+        response = self.client.post(
+            "/api/liveregistrations/push_attendance/",
+            {"live_attendance": {"k2": "v2"}},
+            content_type="application/json",
+            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+        )
+        # New record is set as email is different
```

Comment:
I don't understand this sentence. There is no new record and I see no email in this scenario

Reply:
old vestige thanks